ui/apps/platform/cypress/integration/collections/permissions.test.js

```diff
@@ -0,0 +1,116 @@
+import withAuth from '../../helpers/basicAuth';
+import { hasFeatureFlag } from '../../helpers/features';
+import { visit, visitWithStaticResponseForPermissions } from '../../helpers/visit';
+import navSelectors from '../../selectors/navigation';
+import { tryDeleteCollection, visitCollections } from './Collections.helpers';
+import { collectionSelectors } from './Collections.selectors';
+
+// Quick utility to create a basic collection for permission testing
+function createCollection(name) {
+    // Ignore autocomplete requests
+    // TODO Remove this once the feature is in
+    cy.intercept('/v1/collections/autocomplete', {});
+    localStorage.setItem('access_token', Cypress.env('ROX_AUTH_TOKEN'));
+    tryDeleteCollection(name);
+    visitCollections();
+    cy.get('a:contains("Create collection")').click();
+    cy.get('input[name="name"]').type(name);
+    cy.get('input[name="description"]').type('A collection for permission testing purposes');
+    cy.get('button:contains("All namespaces")').click();
+    cy.get('button:contains("Namespaces with names matching")').click();
+    cy.get('input[aria-label="Select value 1 of 1 for the namespace name"]').type('stackrox');
+    cy.get(`button:contains('stackrox')`).click();
+    cy.get('button:contains("Save")').click();
+}
+
+describe('Collection permission checks', () => {
+    withAuth();
+
+    const collectionName = 'Permission test collection';
+
+    beforeEach(function beforeHook() {
+        if (!hasFeatureFlag('ROX_OBJECT_COLLECTIONS')) {
+            this.skip();
+        }
+    });
+
+    // Ensure a collection exists in the system for permission tests
+    if (hasFeatureFlag('ROX_OBJECT_COLLECTIONS')) {
+        before(() => createCollection(collectionName));
+        after(() => tryDeleteCollection(collectionName));
+    }
```

Comment:
This `before` and `after` idiom makes sense because there are no default collections.

The other side of the coin is (as you have done) for test files to have clear purpose. In the past, imperative idioms like `beforeEach` outside of tests caused confusion in large test files which had no clear focus.

Reply:
Discussion above might remove need for this idiom.

Reply:
Hmm it could, if the first test in the suite runs the creation function and the last test in the suite runs deletion. (Although if we opt to not delete, that isn't necessary.)

I kind of like the `before`/`after` idiom though, when it is fairly minimal like this. It gives a clear location for setup/teardown type tasks.

Reply:
Yeah, good point. Although I prefer explicit function call in each test which requires a (possibly conditionally created) view-only collection, teardown to delete at the end of the test file closes the door on a subtle intermittent failure if cypress occasionally runs test files out or order so (usually) following test file depends on the collection without having called the function.

Reply:
Although so far I have seen only subfolders out of order, cypress makes no guarantee about order of test files.